R/derive_flag_select_msrc.R

```diff
@@ -0,0 +1,267 @@
+#'   by the specified variables for evaluating the condition. If named elements
+#'   are used in `by_vars` like `by_vars = exprs(USUBJID, EXLNKID = ECLNKID)`,
+#'   the variables are renamed after the evaluation. If the `by_vars` element is
+#'   not specified, the observations are grouped by the variables specified for
+#'   the `by_vars` argument.
+#'
+#'   1. The new variable (`new_var`) is added to the input dataset and set to
+#'   the true value (`true_value`) if for the by group at least one condition
+#'   evaluates to `TRUE` in one of the sources. It is set to the false value
+#'   (`false_value`) if for the by group at least one observation exists and for
+#'   all observations the condition evaluates to `FALSE` or `NA`. Otherwise, it
+#'   is set to the missing value (`missing_value`).
+#'
+#' @seealso [flag_event()]
+#'
+#' @family der_gen
+#' @keywords der_gen
+#'
+#' @export
+#'
+#' @examples
+#' library(dplyr)
+#'
+#' adsl <- tribble(
+#'   ~USUBJID,
+#'   "1",
+#'   "2",
+#'   "3",
+#'   "4"
+#' )
+#'
+#' cm <- tribble(
+#'   ~USUBJID, ~CMCAT,        ~CMSEQ,
+#'   "1",      "ANTI-CANCER",      1,
+#'   "1",      "GENERAL",          2,
+#'   "2",      "GENERAL",          1,
+#'   "3",      "ANTI-CANCER",      1
+#' )
+#'
+#' pr <- tibble::tribble(
+#'   ~USUBJID, ~PRSEQ,
+#'   "2",      1,
+#'   "3",      1
+#' )
+#'
+#' derive_flag_select_msrc(
+#'   adsl,
+#'   flag_events = list(
+#'     flag_event(
+#'       dataset_name = "cm",
+#'       condition = CMCAT == "ANTI-CANCER"
+#'     ),
+#'     flag_event(
+#'       dataset_name = "pr"
+#'     )
+#'   ),
+#'   source_datasets = list(cm = cm, pr = pr),
+#'   by_vars = exprs(USUBJID),
+#'   new_var = CANCTRFL
+#' )
+#'
+#' # using different by variables depending on the source
+#' adex <- tribble(
+#'   ~USUBJID, ~EXLNKID, ~EXADJ,
+#'   "1",      "1",      "AE",
+#'   "1",      "2",      NA_character_,
+#'   "1",      "3",      NA_character_,
+#'   "2",      "1",      NA_character_,
+#'   "3",      "1",      NA_character_
+#' )
+#'
+#' ec <- tribble(
+#'   ~USUBJID, ~ECLNKID, ~ECADJ,
+#'   "1",      "3",      "AE",
+#'   "3",      "1",      NA_character_
+#' )
+#'
+#' fa <- tribble(
+#'   ~USUBJID, ~FALNKID, ~FATESTCD, ~FAOBJ,            ~FASTRESC,
+#'   "3",      "1",      "OCCUR",   "DOSE ADJUSTMENT", "Y"
+#' )
+#'
+#' derive_flag_select_msrc(
+#'   adex,
+#'   flag_events = list(
+#'     flag_event(
+#'       dataset_name = "ex",
+#'       condition = !is.na(EXADJ)
+#'     ),
+#'     flag_event(
+#'       dataset_name = "ec",
+#'       condition = !is.na(ECADJ),
+#'       by_vars = exprs(USUBJID, EXLNKID = ECLNKID)
+#'     ),
+#'     flag_event(
+#'       dataset_name = "fa",
+#'       condition = FATESTCD == "OCCUR" & FAOBJ == "DOSE ADJUSTMENT" & FASTRESC == "Y",
+#'       by_vars = exprs(USUBJID, EXLNKID = FALNKID)
+#'     )
+#'   ),
+#'   source_datasets = list(ex = adex, ec = ec, fa = fa),
+#'   by_vars = exprs(USUBJID, EXLNKID),
```

Comment:
can this be moved up to align with the function documentation? 
![image](https://github.com/pharmaverse/admiral/assets/10111024/36f899e4-134d-449a-a2c2-86128a2decb0)

Reply:
Moved